src/common/types/objects/ledger_entries.ts

```diff
@@ -73,11 +156,16 @@ export interface SignerListLedgerEntry {
   PreviousTxnLgrSeq: number
 }
 
-// TODO: Add the other ledger entry types, then remove the `any` fallback
 // see https://ripple.com/build/ledger-format/#ledger-object-types
 export type LedgerEntry =
   AccountRootLedgerEntry |
   AmendmentsLedgerEntry |
+  DirectoryNodeLedgerEntry |
+  OfferDirectoryNodeLedgerEntry |
+  EscrowLedgerEntry |
+  FeeSettingsLedgerEntry |
+  LedgerHashesLedgerEntry |
+  OfferLedgerEntry |
   PayChannelLedgerEntry |
-  SignerListLedgerEntry |
-  any
```

Comment:
Do we want to add `CheckLedgerEntry` here?

Reply:
Good catch Elliot. I added `CheckLedgerEntry` between `AmendmentsLedgerEntry` and `DirectoryNodeLedgerEntry` to keep things in same order as the the documentation (alphabetical, if I remember right). 

Will it be possible to condense all of these commits into one for clarity, or is that something I should do on my side first?

Reply:
I believe I can use GitHub’s squash option to merge this PR as one commit. But if you prefer to do it on your side, that’s fine too. So - either way :)

Reply:
fwiw, `DepositPreauth` is also on the way in rippled 1.1.0 (pending amendment vote)
https://github.com/ripple/rippled/blob/7b0367730cf11b05395f24b537247d7341ea106a/src/ripple/protocol/impl/LedgerFormats.cpp#L172-L178

Reply:
@intelliot I'd say lets use the squash option.

@wilsonianb Thanks for the input. I went ahead and added the DepositPreauth legder_entry. Even if it doesn't get voted in right away, I saw no harm considering the types don't affect runtime performance. All of the fields were easy to interpret due to their use in the other entries except for `Authorize` which I had to pick up from the SFields.cpp file. Can you confirm for me that I interpreted `Authorize` as a string (account) correctly. I've been learning cpp because of the rippled code, and have made my way through most of it, but I am far from proficient. 

https://github.com/ripple/rippled/blob/7b0367730cf11b05395f24b537247d7341ea106a/src/ripple/protocol/impl/SField.cpp#L223

```Javascript
export interface DepositPreauthLedgerEntry {
  LedgerEntryType: 'DepositPreauth',
  Account: string,
  Authorize: string,
  OwnerNode: string,
  PreviousTxnID: string,
  PreviousTxnLgrSeq: number
}
```

Reply:
Nice digging. Yes, `Authorize` is a string.